app/services/catalog/update_order_item.rb

```diff
@@ -17,15 +17,11 @@ def process
       ManageIQ::API::Common::Request.with_request(order_item.context.transform_keys(&:to_sym)) do
         order_item.update_message("info", "Task update message received with payload: #{@payload}")
 
-        if @payload["state"] == "completed"
-          order_item.state = "Order Completed"
-          order_item.update_message("info", "Order Complete")
-
-          order_item.external_url = fetch_external_url
-
-          Rails.logger.info("Updating OrderItem: #{order_item.id} with 'Order Completed' state and #{order_item.external_url}")
-          order_item.save!
-          Rails.logger.info("Finished updating OrderItem: #{order_item.id} with 'Order Completed' state")
+        if @payload["status"] == "ok"
+          mark_item_finished(order_item)
+          order_item.order.finalize_order
+        else
+          mark_item_failed(order_item)
```

Comment:
Is there a need to also mark the order failed as well here?

Reply:
@eclarizio An order comprises of multiple order_items, so once we update a single order_item we should call the order to handle itself, which will cause it to look thru its order_items and depending on the state of each of the order_items it will mark the order accordingly.

Thats what this code does in order
https://github.com/ManageIQ/catalog-api/blob/fd623ddfbd9d2d5dc8793afe2bc147ba8718811e/app/models/order.rb#L19

Reply:
@mkanoor Correct me if I'm wrong, but Isn't that what I'm doing on line 22? I think @syncrou is saying that I should probably move line 22 under the if/else so that it happens regardless.

Reply:
@eclarizio That also needs to be done when an item fails.
Can we break that function into smaller function.
Also there can be 3 updates that could come in here
  - A progress message
 - A successful completion message
- A failed completion message

The code the way it looks right now is checking for status='ok' to mark a successful completion all else is considered failure it doesn't account for the progress messages coming from Openshift.

Also can the order_item be made into an instance variable since each of these messages that we are reacting to is tied to a single order_item so we don't have to pass it between functions